src/main/java/io/zold/api/WalletsIn.java

```diff
@@ -76,12 +80,23 @@ public Wallet create() {
         throw new UnsupportedOperationException("create() not yet supported");
     }
 
-    // @todo #4:30min Read instance of the Wallet from file and put it
-    //  to the result. Should be taken care of after Wallet interface will have
-    //  necessary implementations. Cover with tests and remove irrelevant test
-    //  case.
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public Iterator<Wallet> iterator() {
-        throw new UnsupportedOperationException("iterator() not yet supported");
+        // @checkstyle MagicNumber (1 line)
+        final List<Wallet> wallets = new ArrayList<>(10);
```

Comment:
@carlosmiranda can you use `Directory` and `Filtered` from cactoos here? Also, keep in mind that the default filename extension for wallets is [`.z`](https://github.com/zold-io/zold/pull/282).